libs/dh/market-participant/feature-edit-actor/src/lib/market-participant-edit-actor/master-data/dh-market-participant-actor-master-data.component.ts

```diff
@@ -21,39 +21,115 @@ import {
   templateUrl: './dh-market-participant-actor-master-data.component.html',
 })
-export class DhMarketParticipantActorMasterDataComponent implements OnChanges {
+export class DhMarketParticipantActorMasterDataComponent
+  implements OnChanges, OnDestroy
+{
   @Input() actor: ActorDto | undefined;
   @Output() hasChanges = new EventEmitter<ActorChanges>();
-  changes: ActorChanges = { gln: '', marketRoles: [], meteringPointTypes: [] };
+  changes: ActorChanges = {
+    gln: '',
+    status: 'New',
+    marketRoles: [],
+    meteringPointTypes: [],
+  };
+
+  private destroy$ = new Subject<void>();
+  allStatuses: WattDropdownOption[] = [];
+  statuses: WattDropdownOption[] = [];
+
+  constructor(private translocoService: TranslocoService) {
+    this.translocoService
+      .selectTranslateObject(
+        'marketParticipant.actor.create.masterData.statuses'
+      )
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((statusKeys) => {
+        this.allStatuses = Object.keys(statusKeys)
+          .map((key) => ({
+            value: key,
+            displayValue: statusKeys[key],
+          }))
+          .sort((a, b) => a.displayValue.localeCompare(b.displayValue));
+        this.statuses = this.getValidStatusTransitionOptions(
+          this.actor?.status ?? 'New'
+        );
+      });
+  }
 
   ngOnChanges(): void {
     if (this.actor !== undefined) {
       this.changes = {
         gln: this.actor.gln.value,
         marketRoles: this.actor.marketRoles,
         meteringPointTypes: this.actor.meteringPointTypes,
+        status: this.actor.status,
       };
+      this.statuses = this.getValidStatusTransitionOptions(this.changes.status);
       this.hasChanges.emit({ ...this.changes });
     }
   }
 
+  ngOnDestroy(): void {
+    this.destroy$.next();
+    this.destroy$.unsubscribe();
+  }
+
   readonly onModelChanged = () => {
     this.hasChanges.emit({ ...this.changes });
   };
+
+  private getValidStatusTransitionOptions = (status: ActorStatus) => {
+    switch (status) {
+      case ActorStatus.New:
+        return this.allStatuses.filter((x) =>
+          [
+            ActorStatus.New.toLowerCase(),
+            ActorStatus.Active.toLowerCase(),
+            ActorStatus.Deleted.toLocaleLowerCase(),
+          ].includes(x.value.toLowerCase())
+        );
+      case ActorStatus.Active:
+      case ActorStatus.Inactive:
+      case ActorStatus.Passive:
+        return this.allStatuses.filter((x) =>
+          [
+            ActorStatus.Active.toLowerCase(),
+            ActorStatus.Inactive.toLocaleLowerCase(),
+            ActorStatus.Passive.toLocaleLowerCase(),
+            ActorStatus.Deleted.toLocaleLowerCase(),
+          ].includes(x.value.toLowerCase())
+        );
+      case ActorStatus.Deleted:
+        return this.allStatuses.filter((x) =>
+          [ActorStatus.Deleted.toLocaleLowerCase()].includes(
+            x.value.toLowerCase()
+          )
+        );
+      default:
+        return [];
+    }
```

Comment:
suggestion: looks like returning the correct set of statuses is important. If so, consider extracting the whole function to a new file, say `getValidStatusTransitionOptions.ts`, that takes both the `allStatuses` and `status` as params. This way we can write unit tests on it either as part of the PR or after we the introduction to testing.